sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ResolvedDataSource.scala

```diff
@@ -179,6 +179,13 @@ object ResolvedDataSource extends Logging {
           val fs = path.getFileSystem(sqlContext.sparkContext.hadoopConfiguration)
           path.makeQualified(fs.getUri, fs.getWorkingDirectory)
         }
+
+        partitionColumnsSchema(data.schema, partitionColumns).foreach { field =>
+          if (!PartitioningUtils.validPartitionColumnTypes.contains(field.dataType)) {
+            throw new AnalysisException(s"Cannot use ${field.dataType} for partition column")
+          }
+        }
```

Comment:
ok, I'll make this a function in PartitioningUtils to throw an analysisException.